src/tic_tac_toe/logs/game_to_log.clj

```diff
@@ -0,0 +1,39 @@
+(ns tic-tac-toe.logs.game-to-log
+  (require [tic-tac-toe.ui.user-interface :as ui]
+           [tic-tac-toe.validation.log-validator :as log-validator]))
+
+(defn named-same
+  [game log]
+  (-> (:name log)
+      (nil?)
+      (not)
+      (and (= (:name log) (:name game)))))
+
+(defn associate-name
+  [name game]
+  (->> name
+       (hash-map :name)
+       (merge game)))
+
+(defn associate-id
+  [{:keys [id]} game]
+  (->> id
+       (hash-map :id)
+       (merge game)))
+
+(defn- get-name-selection
+  ([game prompt] (get-name-selection game prompt ""))
+
+  ([{:keys [ui invalid-selection] :as game} prompt error]
+    (let [input (->> prompt
+                     (str error)
+                     (ui/prompt-selection ui))]
+      (if (log-validator/valid-name? input)
+        input
+        (recur game prompt invalid-selection)))))
```

Comment:
Do we need to have both signatures for this function? It appears that the first is the only one that will ever be called.